src/lib.rs

```diff
@@ -0,0 +1,1175 @@
+//
+// The Universal Permissive License (UPL), Version 1.0
+//
+// Subject to the condition set forth below, permission is hereby granted to any person obtaining a
+// copy of this software, associated documentation and/or data (collectively the "Software"), free
+// of charge and under any and all copyright rights in the Software, and any and all patent rights
+// owned or freely licensable by each licensor hereunder covering either (i) the unmodified
+// Software as contributed to or provided by such licensor, or (ii) the Larger Works (as defined
+// below), to deal in both
+//
+// (a) the Software, and
+// (b) any piece of software and/or hardware listed in the lrgrwrks.txt file
+// if one is included with the Software (each a "Larger Work" to which the Software is contributed
+// by such licensors),
+//
+// without restriction, including without limitation the rights to copy, create derivative works
+// of, display, perform, and distribute the Software and make, use, sell, offer for sale, import,
+// export, have made, and have sold the Software and the Larger Work(s), and to sublicense the
+// foregoing rights on either these or other terms.
+//
+// This license is subject to the following condition: The above copyright notice and either this
+// complete permission notice or at a minimum a reference to the UPL must be included in all copies
+// or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+//! A vector of bits ("Vob") is a sequence of bits which exposes a `Vec`-like interface. Whereas
+//! `Vec<bool>` requires 1 byte of storage per bit, `Vob` requires only 1 bit of storage per bit.
+//!
+//! The main documentation for this crate can be found in the [`Vob`](struct.Vob.html) struct.
+
+extern crate num_traits;
+
+use std::cmp::PartialEq;
+use std::fmt;
+use std::fmt::Debug;
+use std::hash::{Hash, Hasher};
+use std::iter::FromIterator;
+use std::mem::size_of;
+use std::ops::{Index, Range};
+use std::slice;
+
+use num_traits::{PrimInt, One, Zero};
+
+
+/// A Vob is a "vector of bits": a sequence of bits which exposes a `Vec`-like interface. Whereas
+/// `Vec<bool>` requires 1 byte of storage per bit, `Vob` requires only 1 bit of storage per bit.
+/// For larger numbers of bits, Vobs can lead to a significant memory decrease and performance
+/// increase.
+///
+/// # Examples
+/// The `vob!` macro makes creating small `Vob`s easy:
+///
+/// ```rust
+/// # #[macro_use] extern crate vob;
+/// # fn main() {
+/// let mut v = vob![true, false, true];
```

Comment:
Could this macro be updated to also support:
```
vob![1, 0, 0 , 0, 1, 1];
```

for example?

(Just a more concise way to make an initialised vob)

Reply:
Not sure. Also not sure I want to think about it now ;) This notation only makes sense for very tiny Vobs anyway IMHO.

Reply:
True, it was just a passing idea. Let's leave it be.